apps/remote_control/lib/lexical/remote_control/search/indexer/module.ex

```diff
@@ -0,0 +1,24 @@
+defmodule Lexical.RemoteControl.Search.Indexer.Module do
+  alias Lexical.RemoteControl.Search.Indexer
+
+  def index(module) do
+    with true <- indexable?(module),
+         {:ok, path, source} <- source_file_path(module) do
+      Indexer.Source.index(path, source)
+    end
+  end
+
+  def source_file_path(module) do
+    with {:ok, %{file: file_path}} <- BeamFile.debug_info(module),
```

Comment:
Playing around with this locally, this `debug_info` call takes quite a long time. If all we care about is the source path and contents, this is available on the module directly using `__info__(:compile)`:

```elixir
iex(1)> Lexical.RemoteControl.__info__(:compile)
[
  version: '8.2.6.3',
  options: [:no_spawn_compiler_process, :from_core, :no_core_prepare,
   :no_auto_import],
  source: '/home/z/dev/lexical/apps/remote_control/lib/lexical/remote_control.ex'
]
```

Would also make `BeamFile` dep unnecessary.

Reply:
fantastic. I was originally using BeamFile as a way to reconstitute the source code from certain beam files, but that didn't pan out (the code it generated was definitely incorrect). I'll swap these out.